packages/perseus-core/src/utils/add-library-version-To-perseus-debug.ts

```diff
@@ -0,0 +1,29 @@
+/**
+ * Adds the given perseus library version information to the __perseus_debug__
+ * object and ensures that the object is attached to `globalThis` (`window` in
+ * browser environments).
+ *
+ * This allows each library to provide runtime version information to assist in
+ * debugging in production environments.
+ */
+export const addLibraryVersionToPerseusDebug = (
+    libraryName: string,
+    libraryVersion: string,
+) => {
+    // If the library version is the default value, then we don't want to
+    // prefix it with a "v" to indicate that it is a version number.
+    let prefix = "v";
+    if (libraryVersion === "__lib_version__") {
+        prefix = "";
+    }
+    const s = `${prefix}${libraryVersion}`;
+
+    if (globalThis) {
+        globalThis.__perseus_debug__ = globalThis.__perseus_debug__ ?? {};
+
+        globalThis.__perseus_debug__[libraryName] = s;
+    } else {
+        // eslint-disable-next-line no-console
+        console.warn(`globalThis not found found (${s})`);
+    }
```

Comment:
**question:** are there environments where this code runs where `globalThis` is not available?

**suggestion:** a more robust check for `globalThis` would be:
```
if (typeof globalThis !== "undefined") {
    ...
} else {
    ...
}
```

Reply:
iOS 12.1 and lower [don't support](https://caniuse.com/?search=globalThis) `globalThis`.  I know that in our main webapp we recently dropped support for Safari/iOS 12. But I think I'm going to keep this in place just so we don't intentionally break things for iOS 12 users.

I'll fix the way I check though, to what you suggest.